### PR TITLE
Comments on Escrow, lending and markets

### PR DESCRIPTION
**Pull request by mkhoury93**

### CODE REVIEW

**Review by mrbaguvix — APPROVED** [review]

**Review by passabilities — APPROVED** [review]

**Review by mrbaguvix — DISMISSED** [review]

File: contracts/cc/creditcard/CreditCard_V1.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
```
Comment (passabilities) [inline contracts/cc/creditcard/CreditCard_V1.sol:1]: should remove file from PR

File: contracts/escrow/escrow/ILoansEscrow.sol [hunk 2]
```diff
@@ -4,12 +4,29 @@ pragma solidity ^0.8.0;
...
 
+    /**
+     * @notice it approves the spender to spend a maximum amount of a respective token from a token address
+     * @param token address of the respective ERC20 token to approve for the spender
+     * @param spender address of the respective spender who is approved by the token contract
+     */
     function setTokenAllowance(address token, address spender) external;
 
+    /**
+     * @notice it allows user to claim their escrow tokens
```
Comment (passabilities) [inline contracts/escrow/escrow/ILoansEscrow.sol:25]: should also mention that only the `ower` (`TellerDiamond`) can make this call on behalf of the users

File: contracts/escrow/escrow/LoansEscrow_V1.sol [hunk 3]
```diff
@@ -54,6 +65,12 @@ contract LoansEscrow_V1 is ILoansEscrow {
         IERC20(token).approve(spender, type(uint256).max);
     }
 
+    /**
+     * @notice it allows user to claim their escrow tokens
```
Comment (passabilities) [inline contracts/escrow/escrow/LoansEscrow_V1.sol:69]: same ^

File: contracts/lending/ttoken/strategies/compound/TTokenCompoundStrategy_1.sol [hunk 4]
```diff
@@ -22,16 +22,26 @@ import "../../storage.sol" as TokenStorage;
...
 contract TTokenCompoundStrategy_1 is RolesMods, TTokenStrategy {
+    /**
+     * @notice it creates the TToken storage and adds it to a hashed slot position
+     */
     function() pure returns (TokenStorage.Store storage)
         private constant tokenStore = TokenStorage.store;
 
+    /**
+     * @notice it creates the Compound storage and adds it to a hashed slot position
+     */
```
Comment (passabilities) [inline contracts/lending/ttoken/strategies/compound/TTokenCompoundStrategy_1.sol:33]: change to 
> It creates a reference to the {TToken | Compound} storage // remove remaining of sentence

File: contracts/market/CreateLoanFacet.sol [hunk 5]
```diff
@@ -268,13 +291,22 @@ library CreateLoanLib {
         );
     }
 
+    /**
+     * @notice creates a new ID for the loan
```
Comment (passabilities) [inline contracts/market/CreateLoanFacet.sol:295]: change from `creates` to `increments`

File: contracts/market/CreateLoanFacet.sol [hunk 6]
```diff
@@ -268,13 +291,22 @@ library CreateLoanLib {
...
+     */
     function newID() internal returns (uint256 id_) {
         Counters.Counter storage counter =
             MarketStorageLib.store().loanIDCounter;
         id_ = Counters.current(counter);
         Counters.increment(counter);
     }
 
+    /**
+     * @notice it creates a new loan escrow for the respective loan to hold it's funds
```
Comment (passabilities) [inline contracts/market/CreateLoanFacet.sol:306]: > it creates a new loan escrow **contract**

File: contracts/market/collateral/CollateralEscrow_V1.sol [hunk 7]
```diff
@@ -20,6 +20,11 @@ contract CollateralEscrow_V1 is OwnableUpgradeable, ICollateralEscrow {
     address public collateralToken;
     bool internal _isWETH;
 
+    /**
+     * @notice it creates a escrow to store collateral tokens
```
Comment (passabilities) [inline contracts/market/collateral/CollateralEscrow_V1.sol:24]: > it **initializes an** escrow

File: contracts/market/collateral/CollateralEscrow_V1.sol [hunk 8]
```diff
@@ -75,6 +96,10 @@ contract CollateralEscrow_V1 is OwnableUpgradeable, ICollateralEscrow {
         supply_ = _balances[loanID];
     }
 
+    /**
+     * @notice it returns the total supply of our collateral token
```
Comment (passabilities) [inline contracts/market/collateral/CollateralEscrow_V1.sol:100]: > it returns the total supply of then collateral token held by the contract

File: contracts/market/collateral/ICollateralEscrow.sol [hunk 9]
```diff
@@ -2,17 +2,42 @@
 pragma solidity ^0.8.0;
 
 interface ICollateralEscrow {
+    /**
+     * @notice it creates a escrow to store collateral tokens
```
Comment (passabilities) [inline contracts/market/collateral/ICollateralEscrow.sol:6]: same from above

File: contracts/market/collateral/ICollateralEscrow.sol [hunk 10]
```diff
@@ -2,17 +2,42 @@
...
 
+    /**
+     * @notice it returns the supply of the respective loan
+     * @param loanID the respective loan ID
+     * @return supply_ the amount in collateral of the respective loan
+     */
     function loanSupply(uint256 loanID) external view returns (uint256 supply_);
 
+    /**
+     * @notice it returns the total supply of our collateral token
```
Comment (passabilities) [inline contracts/market/collateral/ICollateralEscrow.sol:39]: same from above

File: contracts/market/collateral/CollateralEscrow_V1.sol [hunk 11]
```diff
@@ -52,6 +62,12 @@ contract CollateralEscrow_V1 is OwnableUpgradeable, ICollateralEscrow {
         _balances[loanID] += amount;
     }
 
+    /**
+     * @notice it withdraws an amount of tokens in a respective loanID if the caller is the owner
```
Comment (passabilities) [inline contracts/market/collateral/CollateralEscrow_V1.sol:66]: actually only the `TellerDiamond` is able to call this on behalf of the borrower

File: contracts/market/collateral/ICollateralEscrow.sol [hunk 12]
```diff
@@ -2,17 +2,42 @@
...
 
+    /**
+     * @notice it deposits an amount of the respective collateral token into the escrow
+     * @param loanID the ID of the loan
+     * @param amount the amount of collateral tokens to be deposited
+     */
     function deposit(uint256 loanID, uint256 amount) external payable;
 
+    /**
+     * @notice it withdraws an amount of tokens in a respective loanID if the caller is the owner
```
Comment (passabilities) [inline contracts/market/collateral/ICollateralEscrow.sol:20]: same from above

File: contracts/market/libraries/LibConsensus.sol [hunk 13]
```diff
@@ -63,30 +75,41 @@ library LibConsensus {
...
+
+            // check if consensus response has expired
             require(
                 uint256(response.responseTime) >=
                     block.timestamp -
                         PlatformSettingsLib.getTermsExpiryTimeValue(),
                 "Teller: consensus response expired"
             );
+
+            // check if the signature of
```
Comment (passabilities) [inline contracts/market/libraries/LibConsensus.sol:112]: > check if the signature of hashed response data matches

File: contracts/market/libraries/LibLoans.sol [hunk 14]
```diff
@@ -52,6 +52,7 @@ library LibLoans {
      * @return uint256 The total owed amount.
      */
     function getTotalOwed(uint256 loanID) internal view returns (uint256) {
+        //
```
Comment (passabilities) [inline contracts/market/libraries/LibLoans.sol:55]: can remove this but maybe add couple more comments for the functions above

File: contracts/nft/distributor/entry/initialize.sol [hunk 15]
```diff
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: MIT
+ // SPDX-License-Identifier: MIT
```
Comment (passabilities) [inline contracts/nft/distributor/entry/initialize.sol:1]: can undo

**Review by passabilities — APPROVED** [review]

**Review by mrbaguvix — APPROVED** [review]

**Comment by mkhoury93** [comment]
updated from your comments @passabilities !

File: contracts/nft/distributor/interfaces/ITellerNFTDistributor.sol [hunk 16]
```diff
@@ -63,10 +66,23 @@ interface ITellerNFTDistributor {
...
 
+    /**
+     * @notice it removes either an ADMIN or a MINTER role from an account
+     * @param role the role to be granted to an account
+     * @param account the account address
+     */
     function revokeRole(bytes32 role, address account) external;
 
+    /**
+     * @notice
```
Comment (sypeer) [inline contracts/nft/distributor/interfaces/ITellerNFTDistributor.sol:84]: missing un?

**Review by sypeer — CHANGES_REQUESTED** [review]
Capin Memo's first strike! One small fix and looks good

**Comment by mkhoury93** [comment]
@sypeer play ball!

**Review by sypeer — APPROVED** [review]

**Pull request merged**